library/core/src/net/ip_addr.rs

```diff
@@ -1224,7 +1242,13 @@ impl Ipv6Addr {
     #[unstable(feature = "ip_bits", issue = "113744")]
     pub const BITS: u32 = 128;
 
-    /// Converts an IPv6 address into host byte order `u128`.
+    /// Converts an IPv6 address into a `u128` representation using native byte order.
+    ///
+    /// Although IPv6 addresses are big-endian, the `u128` value will use the target platform's
+    /// native byte order. That is, the `u128` value is an integer representation of the IPv6
+    /// address and not an integer interpretation of the IPv6 address's big-endian bitstring. This
+    /// means that the `u128` value `&` with `0xffffffffffffffffffffffffffff0000_u128` will set the
```

Comment:
Same confusion and suggestion as above.
```suggestion
    /// means that the `u128` value masked with `0xffffffffffffffffffffffffffff0000_u128` will set the
```